### PR TITLE
Filter locations without `bin/java` executable

### PR DESCRIPTION
**Pull request by xaviarias**
Fixes: 
 - #22921

### CODE REVIEW

**Comment by xaviarias** [comment]
@bot-gradle test this

**Comment by xaviarias** [comment]
@bot-gradle

**Comment by xaviarias** [comment]
@bot-gradle QF

**Comment by xaviarias** [comment]
@bot-gradle test QF

**Comment by xaviarias** [comment]
@bot-gradle test this

File: subprojects/jvm-services/src/test/groovy/org/gradle/jvm/toolchain/internal/JavaInstallationRegistryTest.groovy [hunk 1]
```diff
@@ -23,10 +23,11 @@ import spock.lang.Specification
 
 class JavaInstallationRegistryTest extends Specification {
 
-    def tempFolder = createTempDir()
```
Comment (ljacomet) [inline subprojects/jvm-services/src/test/groovy/org/gradle/jvm/toolchain/internal/JavaInstallationRegistryTest.groovy:26]: Why move this to all methods?

File: subprojects/docs/src/docs/userguide/jvm/toolchains.adoc [hunk 2]
```diff
@@ -198,6 +198,9 @@ JVM auto-detection knows how to work with:
 
 Among the set of all detected JRE/JDK installations, one will be picked according to the <<sec:precedence,Toolchain Precedence Rules>>.
 
+If you are configuring <<sec:custom_loc>>, note that installations that are non-existing or without a
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/jvm/toolchains.adoc:201]: I would put this in a `NOTE` for better visibility.

Also, this is not only for custom locations, it applies to auto detected ones as well.

**Review by ljacomet — CHANGES_REQUESTED** [review]
Looks good overall.

One question for test and one comment for docs, see below.

File: subprojects/jvm-services/src/test/groovy/org/gradle/jvm/toolchain/internal/JavaInstallationRegistryTest.groovy (hunk 1 above)
Comment (xaviarias) [reply subprojects/jvm-services/src/test/groovy/org/gradle/jvm/toolchain/internal/JavaInstallationRegistryTest.groovy:26]: I think I needed this before my last changes, I can revert it.

**Review by ljacomet — APPROVED** [review]

**Comment by xaviarias** [comment]
@bot-gradle test and merge

**Pull request merged**